src/kusto/azext_kusto/tests/latest/preparers.py

```diff
@@ -55,62 +57,103 @@ def create_resource(self, name, **kwargs):
-                .format(name, self.resource_group_name))
+                self.cli_ctx,
+                'az network vnet delete --name {} --resource-group {}'.format(name, self.resource_group_name))
 
 
 class VnetSubnetPreparer(NoTrafficRecordingPreparer, SingleValueReplacer):
     def __init__(self, name_prefix='clitest.vn',
                  parameter_name='subnet',
-                 resource_group_name=None,
                  resource_group_key=KEY_RESOURCE_GROUP,
-                 vnet_name=None,
                  vnet_key=KEY_VIRTUAL_NETWORK,
                  address_prefixes="11.0.0.0/24",
                  dev_setting_name='AZURE_CLI_TEST_DEV_VNET_SUBNET_NAME',
-                 random_name_length=24, key=KEY_VNET_SUBNET):
+                 key=KEY_VNET_SUBNET):
         if ' ' in name_prefix:
             raise CliTestError(
                 'Error: Space character in name prefix \'%s\'' % name_prefix)
-        super(VnetSubnetPreparer, self).__init__(
-            name_prefix, random_name_length)
+        super(VnetSubnetPreparer, self).__init__(name_prefix, 15)
         self.cli_ctx = get_dummy_cli()
         self.parameter_name = parameter_name
         self.key = key
-        self.resource_group_name = resource_group_name
-        self.resource_group_key = resource_group_key
-        self.vnet_name = vnet_name
-        self.vnet_key = vnet_key
+        self.resource_group = [resource_group_key, None]
+        self.vnet = [vnet_key, None]
         self.address_prefixes = address_prefixes
         self.dev_setting_name = os.environ.get(dev_setting_name, None)
 
-    def create_resource(self, name, **kwargs):
+    def create_resource(self, name, **_):
         if self.dev_setting_name:
             return {self.parameter_name: self.dev_setting_name, }
 
-        if not self.resource_group_name:
-            self.resource_group_name = self.test_class_instance.kwargs.get(
-                self.resource_group_key)
-            if not self.resource_group_name:
+        if not self.resource_group[1]:
+            self.resource_group[1] = self.test_class_instance.kwargs.get(
+                self.resource_group[0])
+            if not self.resource_group[1]:
                 raise CliTestError("Error: No resource group configured!")
-        if not self.vnet_name:
-            self.vnet_name = self.test_class_instance.kwargs.get(self.vnet_key)
-            if not self.vnet_name:
+        if not self.vnet[1]:
+            self.vnet[1] = self.test_class_instance.kwargs.get(self.vnet[0])
+            if not self.vnet[1]:
                 raise CliTestError("Error: No vnet configured!")
 
         self.test_class_instance.kwargs[self.key] = 'default'
         return {self.parameter_name: name}
 
-    def remove_resource(self, name, **kwargs):
+    def remove_resource(self, name, **_):
         pass
+
+
+class VnetNicPreparer(NoTrafficRecordingPreparer, SingleValueReplacer):

```

Comment:
Should we use the preparer in our main repo? I thought we already had these two preparers.

Reply:
@qiaozha

Reply:
I am not sure about that. @changlong-liu Can you confirm?

Reply:
Which version of autorest.az is used here?

Reply:
latest autorest.az don't generate tests/latest/preparers.py for kusto.

Hi @astauben ,
Would you try latest az.   Or it's also OK to just remove tests/latest/preparers.py file from this PR.

Reply:
Removed